### ISSUE TITLE
[anidb2] Metadata only showing for Fairy Tail (2012) but not beyond

### ISSUE THREAD

**Issue by DJScias** [ZeroQI/Hama.bundle#428]
Hello there,

The setup is basically 3 folders:

- `Fairy Tail [anidb2-6662]` with 175 episodes
- `Fairy Tail 2nd Season [anidb2-9980]` with 102 episodes
- `Fairy Tail (2018) [anidb2-13295]` with 51 episodes

Once this runs through Plex, it successfully creates one Plex entry with a total of 328 episodes. Which means the scanner did its job just fine.

However after episode 175, when 2nd Season (2014) starts, the metadata is no longer showing. The metadata is missing until episode 328 (the end).
![image](https://user-images.githubusercontent.com/439655/96531018-034b0600-1289-11eb-9890-6f69ea28996e.png)

Below are all the log files I could find, they 11 in total (apologies for that, but I'd prefer to be thorough).
[Fairy Tail (2018) [anidb2-13295].agent-search.log](https://github.com/ZeroQI/Hama.bundle/files/5405864/Fairy.Tail.2018.anidb2-13295.agent-search.log)
[Fairy Tail (2018) [anidb2-13295].agent-update.log](https://github.com/ZeroQI/Hama.bundle/files/5405865/Fairy.Tail.2018.anidb2-13295.agent-update.log)
[Fairy Tail (2018) [anidb2-13295].filelist.log](https://github.com/ZeroQI/Hama.bundle/files/5405855/Fairy.Tail.2018.anidb2-13295.filelist.log)
[Fairy Tail (2018) [anidb2-13295].scanner.log](https://github.com/ZeroQI/Hama.bundle/files/5405856/Fairy.Tail.2018.anidb2-13295.scanner.log)
[Fairy Tail [anidb2-6662].agent-search.log](https://github.com/ZeroQI/Hama.bundle/files/5405857/Fairy.Tail.anidb2-6662.agent-search.log)
[Fairy Tail [anidb2-6662].filelist.log](https://github.com/ZeroQI/Hama.bundle/files/5405858/Fairy.Tail.anidb2-6662.filelist.log)
[Fairy Tail [anidb2-6662].scanner.log](https://github.com/ZeroQI/Hama.bundle/files/5405859/Fairy.Tail.anidb2-6662.scanner.log)
[Fairy Tail 2nd Season [anidb2-9980].agent-search.log](https://github.com/ZeroQI/Hama.bundle/files/5405860/Fairy.Tail.2nd.Season.anidb2-9980.agent-search.log)
[Fairy Tail 2nd Season [anidb2-9980].agent-update.log](https://github.com/ZeroQI/Hama.bundle/files/5405861/Fairy.Tail.2nd.Season.anidb2-9980.agent-update.log)
[Fairy Tail 2nd Season [anidb2-9980].filelist.log](https://github.com/ZeroQI/Hama.bundle/files/5405862/Fairy.Tail.2nd.Season.anidb2-9980.filelist.log)
[Fairy Tail 2nd Season [anidb2-9980].scanner.log](https://github.com/ZeroQI/Hama.bundle/files/5405863/Fairy.Tail.2nd.Season.anidb2-9980.scanner.log)

The only thing I noticed was in `Fairy Tail 2nd Season [anidb2-9980].agent-update.log` getting spammed with lines like below. Unfortunately I've no clue what that is asking me to do, as I haven't touched any settings in ages.
```
metadata.seasons[ 1].episodes[176]
[!] 'MyAnimeList' source not in MetaSources dict, please Check case and spelling
metadata.seasons[ 1].episodes[177]
[!] 'MyAnimeList' source not in MetaSources dict, please Check case and spelling
metadata.seasons[ 1].episodes[178]
[!] 'MyAnimeList' source not in MetaSources dict, please Check case and spelling
metadata.seasons[ 1].episodes[179]
[!] 'MyAnimeList' source not in MetaSources dict, please Check case and spelling
......
```

Note: I have tried to manually "Refresh Metadata" once I noticed it was no longer doing anything but the result was the same.

If there's anything else I can provide/have missed, please do let me know.

**Comment by ZeroQI** [comment]
Notes
- "defaulttvdbseason='a'" in mapping file so ASS used absolute numbering
- should be season 5 if folloIng tvdb numbering
- Scanner code to be altered to use tvdb4 mode instead of just tvdb for metadata to match but season 0 eps to check whether to use tvdb4 or tvdb by checking s1e1 eps if they are in abs mode or TheTVDB season mode? Too complex for now...
- Easiest is to support absolute numbering in tvdb mode if eps after first season are there in first season and no eps in further seasons and  defaulttvdbseason='a'... We could do a check to enable tvdb4 mode in place of tvdb mode if these conditions are in place, hence minimum code change

**Comment by DJScias** [comment]
Hello there,

Is there anything else from my side I need to present, or is there enough info for the issue to be pinpointed?

**Comment by ronia-dev** [comment]
> Hello there,
> 
> Is there anything else from my side I need to present, or is there enough info for the issue to be pinpointed?

I was just coming here to file this issue and found that you already had one.  If you want a quick/dirty fix to hold you over in the meantime, you can edit the file 'Plex Media Server/Plug-in Support/Data/com.plex.app.agents.hama/DataItems/AnimeLists/anime-list-corrections.xml

and add these entries:
```  
  <anime anidbid="9980" tvdbid="114801" defaulttvdbseason="5" episodeoffset="" tmdbid="" imdbid="">
    <name>Fairy Tail (2014)</name>
  </anime>
  <anime anidbid="13295" tvdbid="114801" defaulttvdbseason="8" episodeoffset="" tmdbid="" imdbid="">
    <name>Fairy Tail (2018)</name>
  </anime>
```


I created a similar issue over on scudlee's repo here as well: https://github.com/ScudLee/anime-lists/issues/364

**Comment by sven-7** [comment]
Did this correction work?

**Comment by ronia-dev** [comment]
It does for me.  Don't know if it worked for OP.

Also, don't know what will happen in a couple weeks since it sounds like this is a cached version of the file that is updated once every two weeks.  So it might be overwritten and have to be applied manually again.

**Comment by DJScias** [comment]
Well, I'm willing to try out this fix unless it's better I ask for it to be fixed upstream?

What would you have me do? @ZeroQI @sven-7 @ronia-dev

**Comment by sven-7** [comment]
I wonder if this is broken because you're using TVDB2? I use TVDB3 and I have full metadata for Fairy Tail. It breaks it up into the corresponding 8 TVDB seasons. I'm scared to move "a" out of defaulttvdbseason.

**Comment by DJScias** [comment]
Well, I'm not using TVDB2. I'm using ANIDB2.

The reason why I chose ANIDB2 over using any of the TVDB is because all the TVBD options I tried made it even more of a mess. This was the only one that really gave me a "single season full list" that worked, aside from the obvious issue that I made this post for (metadata not showing for episode 176 and above).

**Comment by sven-7** [comment]
This should be updated now in the new anime-lists repo: https://github.com/Anime-Lists/anime-lists

If you're updated to the latest ASS/HAMA, you should get this list. Please submit an issue there/here if that does not fix it.

**Comment by DJScias** [comment]
Hello there.

I've updated ASS/HAMA but I had to rename `AnimeLists` folder as the old one was still cached.

Once I did that and re-added Fairy Tail it now had 3 season: Season 1, Season 5 and Season 8.

So I guess it worked for ANIDB2's purposes? I've now switched over to TVDB5's version though as I preferred a single list from ep 1 to 328 in one "season". Previously ANIDB2 did this in its bugged state (as it wasn't making new season folders) but with that fixed I had to switch. But now with TVDB5 it just pulled all metadata after a bit of renaming on file numbers.

I suppose this means the issue is solved? I'm not sure myself if that's how it should be (the three folders like that). But if it is, feel free to close it.
